rest/data/host_create_test.go

```diff
@@ -163,13 +177,8 @@ buildvariants:
 		pp.Id = "v1"
 		assert.NoError(t, pp.Insert())
 
-		settings := &evergreen.Settings{
-			Credentials: map[string]string{"github": "token globalGitHubOauthToken"},
-		}
-		assert.NoError(t, evergreen.UpdateConfig(settings))
-
-		assert.NoError(t, CreateHostsFromTask(ctx, settings, &t1, user.DBUser{Id: "me"}, ""))
-		createdHosts, err := host.Find(host.IsUninitialized)
+		assert.NoError(t, CreateHostsFromTask(ctx, env, &t1, user.DBUser{Id: "me"}, ""))
+		createdHosts, err := host.Find(db.Query(bson.M{host.StartedByKey: "me"}))
```

Comment:
Is there a race with the provisioning-create-host task here? This query will only make sense if that task has run, right?

Reply:
I think there is a race (so `host.IsUninitialized` was inconsistent) but the `StartedBy` field is set [when the host intent is created](https://github.com/evergreen-ci/evergreen/blob/88bc52da38ee865940f567a33838051603be1c87/model/host/host_intent.go#L60) and persisted when it's started, so it seemed like it would cover both cases.